akka-distributed-data/src/main/scala/akka/cluster/ddata/GCounter.scala

```diff
@@ -39,12 +39,14 @@ object GCounter {
  */
 @SerialVersionUID(1L)
 final class GCounter private[akka] (
-  private[akka] val state: Map[UniqueAddress, BigInt] = Map.empty)
-  extends ReplicatedData with ReplicatedDataSerialization with RemovedNodePruning with FastMerge {
+  private[akka] val state: Map[UniqueAddress, BigInt] = Map.empty,
+  override val delta:      Option[GCounter]           = None)
```

Comment:
I am going back and forth with the choice of linking the last delta to the current state.

**Advantages**:
- Your mutation operations can directly apply the delta they would otherwise compute, instead of mutation + merge the resulting delta. I'm not certain whether this provides significant speed benefits though.
- Some complexity can be offloaded elsewhere instead of in the Replicator, which in my eyes is another candidate for the delta-state management.

**Disadvantages**
- You cannot have multiple deltas with your current state. Ideally, the delta is the exact difference between two replicas. Depending on the protocol (many possibilities here), the difference may be replica-pair-specific. This choice therefore seems to eliminate possible protocol implementations.
- Each delta-CRDT mutation must effectively implement merge, or fall back to computing the delta and then calling merge, which is not DRY.